pkg/cri/resource-manager/requests.go

```diff
@@ -204,6 +229,18 @@ func (m *resmgr) CreateContainer(ctx context.Context, method string, request int
 	m.Lock()
 	defer m.Unlock()
 
+	// kubelet doesn't always clean up crashed containers so we try doing it here
+	if msg, ok := request.(*criapi.CreateContainerRequest); ok {
+		if pod, ok := m.cache.LookupPod(msg.PodSandboxId); ok {
+			if msg.Config != nil && msg.Config.Metadata != nil {
+				if c, ok := pod.GetContainer(msg.Config.Metadata.Name); ok {
+					m.Warn("re-creation of container %s, releasing old one", c.PrettyName())
+					m.policy.ReleaseResources(c)
```

Comment:
I don't maybe get the logic here... are the stale resources released both in `CreateContainer()` and `RunPod()`? Wouldn't one place be enough?

Reply:
> I don't maybe get the logic here... are the stale resources released both in `CreateContainer()` and `RunPod()`? Wouldn't one place be enough?

Those are cleanups for different objects: 1) a lingering pod (with any potential lingering containers), 2) a single lingering container.